server/interceptor/interceptor.go

```diff
@@ -229,129 +221,51 @@ func isLeafOK(leaf *trillian.QueuedLogLeaf) bool {
 	return leaf == nil || leaf.Status == nil || leaf.Status.Code == int32(codes.OK)
 }
 
-// rpcInfo contains information about an RPC, as extracted from its request message.
 type rpcInfo struct {
-	doNotIntercept bool
+	// getTree and quota enable their corresponding interceptor logic.
+	getTree, quota bool
 
-	// treeID is the tree ID tied to this RPC, if any (zero means no tree).
-	treeID int64
+	readonly bool
+	treeID   int64
+	treeType trillian.TreeType
 
-	// opts is the trees.GetOpts appropriate to this RPC (TreeType, readonly vs readwrite, etc).
-	// opts is not set if doNotIntercept is true.
-	opts trees.GetOpts
-
-	// specs contains the quota specifications for this RPC.
-	specs []quota.Spec
-
-	// tokens is number of quota tokens consumed by this request.
+	specs  []quota.Spec
 	tokens int
 }
 
-// getRPCInfo returns the rpcInfo for the given request, or an error if the request is not mapped.
-// RPCs are mapped using the following logic:
-// treeID is acquired via one of the "Request" interfaces defined below (treeIDRequest,
-// logIDRequest, mapIDRequest, etc). Requests must implement to one of those.
-// TreeType and Readonly are determined based on the request type.
-func getRPCInfo(req interface{}, quotaUser string) (*rpcInfo, error) {
+func newRPCInfo(req interface{}, quotaUser string) (*rpcInfo, error) {
+	// Set "safe" defaults: enable all interception and assume requests are readonly.
```

Comment:
I think this might be easier to follow if each chunk explicitly set all of the flags -- I found myself paging up and down to double check what the net result would be.

Reply:
I played around with this, but I find the current arrangement better, as it allows us to specify "safe" defaults and have each case focus only on what makes it deviate from (ie, disable X and Y logic, this is readwrite, etc).

Reply:
I'd have thought that not-doing-anything is the safe default (ie. `quota:false, getTree:false`)...

Reply:
It depends on your point of view, I think. Not doing anything is safer to get new things working, but it lets you do stuff such as skip tree state validation, not charge quota, skip auth (when we have it), etc. I prefer to err on the other side: most requests do want those things done, so I enable them by default.

I could have named some boolean fields in the negative, such as noAuth, noQuota, so on, but I find that a bit weird.